src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -99,8 +99,12 @@ class Definitions {
   lazy val RootPackage: TermSymbol = ctx.newSymbol(
     NoSymbol, nme.ROOTPKG, PackageCreationFlags, TypeRef(NoPrefix, RootClass))
 
-  lazy val EmptyPackageClass = ctx.newCompletePackageSymbol(RootClass, nme.EMPTY_PACKAGE).moduleClass.asClass
-  lazy val EmptyPackageVal = EmptyPackageClass.sourceModule.entered
+  lazy val EmptyPackageVal = ctx.newCompletePackageSymbol(RootClass, nme.EMPTY_PACKAGE).entered
+  lazy val EmptyPackageClass = EmptyPackageVal.moduleClass.asClass
+
+  /** A package in which we can place all methods that are interpreted specially by the compiler */
```

Comment:
You're right, `throw` could be a non-inlined method call to JS. But it would require *some* "library" JS to be shipped with every Scala.js output, though, which would not be nice. Of course we have built-in JS ([scalajsenv.js](https://github.com/scala-js/scala-js/blob/master/tools/scalajsenv.js)), but these are part of the the lambda surrounding the Scala.js code, and hence do not leak in the outer JS scope. That `throw` function would need to be leaked in order to be callable from Scala.js code (ironically).